backend/src/ml_space_lambda/initial_app_config/lambda_function.py

```diff
@@ -44,5 +45,30 @@ def lambda_handler(event, context):
     ]
 
     app_configuration_dao.update(config)
+    update_dynamic_roles()
 
     generate_html_response(200, "Successfully updated app config")
+
+
+def update_dynamic_roles():
```

Comment:
This may work for your allowed instances policies, but will not work for allowed services policies as the policy may not be created yet. We probably need to make this an iam_manager function and then it can be called here and when creating the deny services policy

Reply:
Can you expand on why this won't work with the service policies?

Reply:
The services policy doesn't exist on deployment

Reply:
If we end up creating a group policy too then this will also be relevant there. We will want/need to apply this for dynamically created policies

Reply:
I'll can update the name as this is a bit too generic but this is just for updating _existing_ dynamic user roles that were created **before** the instance type constraint policies existed on the notebook role. It should only run once the first time 1.5 is deployed.

Any dynamic roles created after 1.5 is deployed will be tagged appropriately and will have the correct notebook policies.

Reply:
What I am saying is that we need this same logic for the deny role the first time it is created and for any new policies we may create in the future. This is reusable if you put it in the `iam_manager`. If you just want to commit this, then I can do the work to move it later, but I also need this applied to the service policy or otherwise customers will need to manually attach that.

Reply:
I still feel pretty strongly about moving this to the `iam_manager`, but if you don't do it, then I can do it in a separate PR. We need to update existing dynamic roles with the deny services policy too, so this is something that is needed elsewhere.